ydr/shader_materials.py

```diff
@@ -226,21 +226,29 @@ def create_tinted_shader_graph(obj: bpy.types.Object):
         # Blender still has space to append '.012' in case of duplicated names.
         tint_color_attr_name = tint_color_attr_name[:64-4]
 
-        tint_color_attr = obj.data.attributes.new(name=tint_color_attr_name, type="BYTE_COLOR", domain="CORNER")
+        if tint_color_attr_name in obj.data.attributes:
+            tint_color_attr = obj.data.attributes[tint_color_attr_name]
+        else:
+            tint_color_attr = obj.data.attributes.new(name=tint_color_attr_name, type="BYTE_COLOR", domain="CORNER")
 
         rename_tint_attr_node(mat.node_tree, name=tint_color_attr.name)
 
-        create_tint_geom_modifier(obj, tint_color_attr.name, input_color_attr_name, palette_img)
+        modifier_name = f"TintGeometry_{tint_color_attr_name}"
+        existing_modifier = obj.modifiers.get(modifier_name)
+        
+        if existing_modifier is None:
+            create_tint_geom_modifier(obj, tint_color_attr.name, input_color_attr_name, palette_img, modifier_name)
```

Comment:
Modifier names have a length limit so with long texture names the modifier name will get truncated, so this existing modifier check won't find it and still create a duplicated modifier.

This will also still create a duplicated modifier if you for example change the palette texture. Figured the point of this button is for when these modifiers get out-of-sync with the shaders on the object (e.g. deleting tint material, changing palette texture).

The simplest approach I think is to just delete all the tint modifiers (before calling `create_tinted_shader_graph` probably) and then just let `create_tinted_shader_graph` re-create them from scratch again.